compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -606,7 +606,7 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
     // with placeholders, which imply nothing about outlives bounds, and then
     // prove below that the hidden types are well formed.
     let universe = infcx.create_next_universe();
-    let mut idx = 0;
+    let mut idx = ty::BoundVar::ZERO;
```

Comment:
oh fun, I didn't realize newtype indices supported `AddAssign<usize>` or impls for other basic int types

Reply:
Well, it was just added in the followed-up PR in the description; but these newtypes impl'd `Add<usize>` already so it makes sense to have had it in the first place.

Reply:
Yep.
I think they may also benefit from `Add<Self>` and `AddAssign<Self>`, but adding `AddAssign<usize>` was easier to justify (since `Add<usize>` was already impl'd, as @compiler-errors pointed out)

Reply:
I thought I had a good example for a use-case for `Add<Self>`, but I might have imagined it, and on second thought - yeah, doesn't make a lot of sense to have that impl. Not that I added it, anyway 😄